src/main/java/org/springframework/samples/petclinic/coverage/BasicConstructionsController.java

```diff
@@ -0,0 +1,155 @@
+                break;
+        }
+        return returnedText;
+    }
+
+    @GetMapping("/for/{value}")
+    public String forLoop(@PathVariable("value") String value) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < (2 + Math.random() * 10); i++) {
+            stringBuilder.append(value).append(" ");
+        }
+        return stringBuilder.toString();
+    }
+
+    @GetMapping("/lambda/{value}")
+    public String lambda(@PathVariable("value") String value) {
+        TestInterface testInterface;
+        testInterface = (str -> str + " from lambda method");
+        return testInterface.inputString(value);
+    }
+
+    @GetMapping("/lambdaRef/{value}")
+    public String lambdaReference(@PathVariable("value") String value) {
+        TestInterface testInterface = TestClassForReference::inputString;
+        return testInterface.inputString(value);
+    }
+
+    @GetMapping("/while/{value}")
+    public String whileLoop(@PathVariable("value") String value) {
+        StringBuilder stringBuilder = new StringBuilder();
+        do {
+            stringBuilder.append(value).append(" ");
+        } while ((2 + Math.random() * 10) > 5);
+        return stringBuilder.toString();
+    }
+
+    @GetMapping("/labels/{value}")
+    public String labels(@PathVariable("value") String value) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < (2 + Math.random() * 10); i++) {
+            if (i == 3) {
+                continue;
+            }
+            stringBuilder.append(i).append(" ").append(value).append("\n");
+            if (i == 9) {
+                break;
+            }
+        }
+        return stringBuilder.toString();
+    }
+
+    @GetMapping("/stream/{value}")
+    public String stream(@PathVariable("value") String value) {
+        List<String> listOfPlanets = new ArrayList<>();
+        listOfPlanets.add("Mercury");
+        listOfPlanets.add("Venus");
+        listOfPlanets.add("Earth");
+        listOfPlanets.add("earth");
+        listOfPlanets.add("Mars");
+        listOfPlanets.add("Jupiter");
+        listOfPlanets.add("Saturn");
+        listOfPlanets.add("saTURN");
+        listOfPlanets.add("Uranus");
+        listOfPlanets.add("Neptune");
+
+        return listOfPlanets.stream().filter((x) -> x.equalsIgnoreCase(value)).map(String::toUpperCase).collect(Collectors.joining(" "));
+    }
+
+    @GetMapping("/trycatch/{value}")
+    public String tryCatch(@PathVariable("value") String value) {
+        StringBuilder stringBuilder = new StringBuilder();
+        String exceptionMessage = "";
+        try {
+            Integer.parseInt(value);
+        } catch (NumberFormatException e) {
+            exceptionMessage = e.getMessage();
+        } finally {
+            stringBuilder.append(value).append(" ").append(exceptionMessage);
+        }
+        return stringBuilder.toString();
+    }
+
+    @GetMapping("/complex/{value}")
```

Comment:
you use here only if-else constructions so it can be:
```suggestion
    @GetMapping("/branches/complex/{value}")
```

Reply:
fised